cpp/include/cuspatial/detail/utility/zero_data.cuh

```diff
@@ -0,0 +1,44 @@
+
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/traits.hpp>
+
+namespace cuspatial {
+namespace detail {
+
+/**
+ * @brief Zero set a data range on device
+ *
+ * @tparam Iterator type of iterator to the range
+ * @param begin Start iterator to the range
+ * @param end End iterator to the range
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ */
+template <typename Iterator>
+void zero_data(Iterator begin, Iterator end, rmm::cuda_stream_view stream)
```

Comment:
This function should have `async` in the name, I think.  Also consider calling it `zero_range_async`, but I'm fine with either.

```suggestion
void zero_data_async(Iterator begin, Iterator end, rmm::cuda_stream_view stream)
```